.github/workflows/check-build.yml

```diff
@@ -0,0 +1,321 @@
+          echo "base_sha=$(git rev-parse ${{ inputs.comparison_base_ref }})" >>$GITHUB_ENV
+          echo "target_ref=${{ github.ref }}" >>$GITHUB_ENV
+          echo "Building ${{ github.ref }} and comparing against ${{ inputs.comparison_base_ref }}"
+
+      - name: Compute tooling args
+        run: |
+          # TODO: We naively assume that this is a 3.0+ version that will have a daily build.
+          # To support older versions, some work may be required.
+          echo 'azl_extra_make_args=DAILY_BUILD_ID=lkg' >>$GITHUB_ENV
+
+      - name: Infer changed specs
+        run: |
+          updated_specs=$(./toolkit/scripts/detect_changes.py --since ${{ env.base_sha }} --spec-names)
+          updated_extended_specs=$(./toolkit/scripts/detect_changes.py --since ${{ env.base_sha }} --extended-spec-names)
+          echo "Specs updated: '${updated_specs}'"
+          echo "Extended specs updated: '${updated_extended_specs}'"
+          echo "updated_specs=$(echo ${updated_specs})" >>$GITHUB_ENV
+          echo "updated_extended_specs=$(echo ${updated_extended_specs})" >>$GITHUB_ENV
+
+      - name: Export outputs
+        id: export
+        run: |
+          echo "UPDATED_SPECS=${{ env.updated_specs }}" >>$GITHUB_OUTPUT
+          echo "UPDATED_EXTENDED_SPECS=${{ env.updated_extended_specs }}" >>$GITHUB_OUTPUT
+          echo "TARGET_REF=${{ env.target_ref }}" >>$GITHUB_OUTPUT
+          echo "AZL_EXTRA_MAKE_ARGS=${{ env.azl_extra_make_args }}" >>$GITHUB_OUTPUT
+
+  build:
+    name: 'Build RPMS'
+    runs-on: ubuntu-latest
+    timeout-minutes: 240
+    needs: [detect-changes]
+    if: ${{ needs.detect-changes.outputs.updated_specs != '' || needs.detect-changes.outputs.updated_extended_specs != '' }}
+    env:
+      updated_specs: ${{ needs.detect-changes.outputs.updated_specs }}
+      updated_extended_specs: ${{ needs.detect-changes.outputs.updated_extended_specs }}
+      azl_extra_make_args: ${{ needs.detect-changes.outputs.azl_extra_make_args }}
+
+    outputs:
+      daily_build_id: ${{ steps.find-daily-repo.outputs.DAILY_BUILD_ID }}
+      daily_build_arch: ${{ steps.find-daily-repo.outputs.DAILY_BUILD_ARCH }}
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Setup Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install Python dependencies for tools
+        run: python3 -m pip install -r toolkit/scripts/requirements.txt
+
+      - name: Find daily repo
+        id: find-daily-repo
+        run: |
+          DAILY_BUILD_ID=$(toolkit/scripts/get_lkg_id.sh /tmp)
+          echo "DAILY_BUILD_ID=${DAILY_BUILD_ID}" >>$GITHUB_OUTPUT
+          echo "DAILY_BUILD_ARCH=x86-64" >>$GITHUB_OUTPUT
+
+      - name: "Build modified specs (base)"
+        id: build-specs
+        if: ${{ env.updated_specs != '' }}
+        run: |
+          sudo make -C toolkit \
```

Comment:
@reubeno Does This workflow assume that the source-tarball is already uploaded to the source cache?

Reply:
Yes, it does. Per our previous discussion that will need to be addressed in follow-up work.